Feature vs3 esp32 i2 c class

RoxanneMango opened the pull request:
- Added I2cEsp class with dummy functions for write() and print()
- Already populated constructor and destructor
- Added member variables such as tx/rx buffer
- Implemented read() function

RoxanneMango committed:
- Added I2C_ESP build map and build config files to .gitignore
- Added I2cEsp class
- Added I2C master-slave initialisation functionality to I2cEsp class
- Added I2C master-slave read functionality to I2cEsp class
- Added I2cEsp class member variables
- Removed SPI code from main.cpp as it's no longer used anyway
- Added master and slave I2cEsp objects to main

RoxanneMango committed:
- Removed stray  #include "wrap-hwlib.hpp" from test_main.cpp as it's not compatible

RoxanneMango committed:
- Updated .clang_format so hwlib.hpp can be found again

RoxanneMango committed:
- Directly modified .fetch-build-environment.sh with most recent git changes from build-environment wiki

RoxanneMango committed:
Made memory_map.hpp conform clang-format

RoxanneMango committed:
- Used clang-format-6.0 on memory_map.hpp

RoxanneMango committed:
- Added Doxygen documentation to I2cEsp class functions

NielsdeWaal commented on ESP/I2C_ESP32/main/I2C_ESP.hpp:
```diff
@@ -0,0 +1,91 @@
+	
+	uint8_t * dataBuffer;	///< Data buffer
+	uint8_t * txBuffer;		///< Tranmission buffer
+	uint8_t * rxBuffer;		///< Receive buffer
```
Raw pointers are reasonable dangerous, try using STL pointers.

RoxanneMango replied:
I2C ESP32 IDF library functions only accept raw pointers

NielsdeWaal commented on ESP/I2C_ESP32/main/I2C_ESP.hpp:
```diff
@@ -0,0 +1,91 @@
+	bool checkAck;			///< Whether to check for ACKs
+	
+	int masterClockFrequency = 100000;
+	uint8_t slaveAddress = 0x28;
```
Both can be constexpr.

RoxanneMango replied:
Turned them into static constexpr :+1:

NielsdeWaal commented on ESP/I2C_ESP32/main/main.cpp:
```diff
@@ -0,0 +1,22 @@
+
+void app_main() {
+    auto led = LED();
+    led.blinkInfinitely(1000);
```
Program will stop at this point, it is an infinite loop.

RoxanneMango replied:
LED code has been commented out for the time being.

NielsdeWaal commented on .clang-format:
```diff
@@ -1,6 +1,4 @@
 ---
-AlignConsecutiveAssignments: true
```
This line should be added back into the file.

RoxanneMango replied:
Added :ok_hand:

NielsdeWaal commented on ESP/I2C_ESP32/main/I2C_ESP.cpp:
```diff
@@ -0,0 +1,70 @@
+	i2c_param_config(portNum, &config);
+	i2c_driver_install(portNum, config.mode, rxBufferLength, txBufferLength, 0);
+	
+	dataBuffer = new uint8_t[dataBufferLength];
```
Maybe don't use ```new``` in the constructor. Try moving it to class header, also see if std::array can be used, for more safe programming.

RoxanneMango replied:
Array has been tried, but would cause problems with buffers

NielsdeWaal reviewed (changes requested):
Try looking at these points.

RoxanneMango committed:
- Changed masterClockFrequency I2cEsp member variable into static constexpr
- Changed slaveAddress I2xEsp member variable into static constexpr
- Changed constexpr variables at top of I2C_ESP.hpp file into static constexpr
- Modified .clang-format so travis won't fail
- Commented out LED code in main.cpp as it was an infinite loop preventing I2C
  code from running

RoxanneMango committed:
- Removed #include <memory> which was a residue of trying to swap raw pointers for smart pointers
- Clang-formatted memory_map.hpp again...
